src/components/tools/edit/EditTool.js

```diff
@@ -136,193 +148,273 @@ function EditToolDialog(props) {
         }
     );
 
+    const confirmOverwrite = (apps) => {
+        setAppsAffectedByTool(apps);
+        setShowOverwriteWarningDialog(true);
+    };
+
     const { mutate: updateCurrentTool, status: updateToolStatus } = useMutation(
-        ({ submission }) =>
-            isAdmin ? adminUpdateTool(submission) : updateTool(submission),
+        () =>
+            isAdmin
+                ? adminUpdateTool(toolSubmission, overwriteAppsAffectedByTool)
+                : updateTool(toolSubmission),
         {
             onSuccess: (data) => {
                 announce({
                     text: t("toolUpdated"),
                 });
                 queryClient.invalidateQueries(TOOLS_QUERY_KEY);
                 setUpdateToolError(null);
+                setOverwriteAppsAffectedByTool(false);
+                setAppsAffectedByTool(null);
+                setToolSubmission(null);
                 onClose();
             },
-            onError: setUpdateToolError,
+            onError: (err) => {
+                console.error(getErrorCode(err));
```

Comment:
Missed this last review, but this looks like debug logging that can be removed now.